src/types/messages/message.ts

```diff
@@ -82,4 +83,6 @@ export interface Message {
   referencedMessage?: Message;
   /** Sent if the message is a response to an Interaction */
   interaction?: MessageInteraction;
+  /** The components related to this message */
+  components: MessageComponents;
```

Comment:
Is it always defined?

Reply:
There is no official docs on this yet so not 100% but yes i believe itll b an empty array